app/lib/katello/resources/cdn.rb

```diff
@@ -66,7 +67,10 @@ def http_downloader
           # Run the following command in rails console to figure out other
           # valid constants in other ruby versions
           # "OpenSSL::SSL::SSLContext::METHODS"
-          net.ssl_version = SETTINGS[:katello][:cdn_ssl_version] if SETTINGS[:katello].key?(:cdn_ssl_version)
+          if Setting[:cdn_ssl_version] && !SUPPORTED_SSL_VERSIONS.include?(Setting[:cdn_ssl_version])
```

Comment:
This is working well, but I think it would be better to move the check to the initialize method in this class so we can bail out as soon as possible. Example:

```
          @ssl_version = Setting[:cdn_ssl_version]                                                                                                                                            
          if @ssl_version && !SUPPORTED_SSL_VERSIONS.include?(@ssl_version)                                                                                                                   
            fail("Invalid SSL version specified. Check the 'CDN SSL Version' setting")                                                                                                        
          end 
```

Then in this method we can set the value:

```
net.ssl_version = @ssl_version
```